src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -47,9 +47,12 @@ public function ask(InputInterface $input, OutputInterface $output, Question $qu
 
         if (!$input->isInteractive()) {
             if ($question instanceof ChoiceQuestion) {
-                $choices = $question->getChoices();
+                $default = $question->getDefault();
+                if (null === $default) {
+                    return $question->isMultiselect() ? array() : null;
+                }
 
-                return $choices[$question->getDefault()];
+                return $question->getValidator() ? $question->getValidator()($default) : $default;
```

Comment:
The validator wasn't applied here previously, this looks like a behavior change to me so cannot be part of the patch. Should be submitted as a new feature on master separately.